notebooks/Functions_EDA_CMM.py

```diff
@@ -0,0 +1,40 @@
+import pandas as pd
+import numpy as np
+
+
+# Define function for generating dataframe which calculates basketsizes per order.
+
+def calc_basketsize (purchases):
+    """Function to generate dataframe with basketsizes out of dataframe with single purchases (e.g. dataframe from transaction_train.csv). 
+    Assumption: Purchases of an individual customer on one day form an order.
+
+    Args:
+        purchases (_dataframe_): Dataframe which contains single purchases per customer in each row
+    """    
+    
+    purchases['datetime'] = pd.to_datetime(purchases['t_dat'])
```

Comment:
Würde ich außerhalb der funktion machen, spart Zeit und ist auch nicht jedes mal nötig.  
Du kannst es auch gleich beim read_csv machen 

`
trans= pd.read_csv("../data/transactions_train.csv",parse_dates=['t_dat']).sort_values("customer_id").reset_index(drop=True)
`
[pandas docu (unübersichtlich)](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html)

[Dataparsing](https://towardsdatascience.com/4-tricks-you-should-know-to-parse-date-columns-with-pandas-read-csv-27355bb2ad0e)

Reply:
Guter Tipp. Im vorliegenden Fall würde ich die datetime-transformation in der Funktion lassen, damit man den Transaction-Datensatz ohne weitere Transformationen direkt in die Funktion werfen kann.

Eigentlich braucht man die Transformation auch gar nicht, habe ich bemerkt. Man könnte groupby auch auf 't_dat' durchführen. Da die Exekution der Funktion aber nicht so lange dauert, kann man die datetime-transformation auch gerade in der Funktion mit erledigen.